packages/core/src/utils-hoist/supports.ts

```diff
@@ -61,7 +61,7 @@ export function supportsDOMException(): boolean {
  * @returns Answer to the given question.
  */
 export function supportsHistory(): boolean {
-  return 'history' in WINDOW;
+  return 'history' in WINDOW && !!WINDOW.history;
```

Comment:
Improved this check so that if history was `undefined` or `null` we'd also return `false` here